cpu/cc2538/dev/ecc-driver.h

```diff
@@ -0,0 +1,227 @@
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+ * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+ * STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED
+ * OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+/**
+ * \addtogroup cc2538-pka
+ * @{
+ *
+ * \defgroup cc2538-ecc cc2538 ECC driver
+ *
+ * Driver for the cc2538 ECC mode of the PKC engine
+ * @{
+ *
+ * \file
+ * Header file for the cc2538 ECC driver
+ */
+#ifndef ECC_DRIVER_H_
+#define ECC_DRIVER_H_
+
+#include "contiki.h"
+#include "pka.h"
+
+#include <stdint.h>
+/*---------------------------------------------------------------------------*/
+/** \name ECC structures
+ * @{
+ */
+typedef struct {
+  const char     *name;       /**< Name of the curve. */
+  const uint8_t  ui8Size;     /**< Size of the curve in 32-bit word. */
+  const uint32_t *pui32Prime; /**< The prime that defines the field of the curve. */
+  const uint32_t *pui32N;     /**< Order of the curve. */
+  const uint32_t *pui32A;     /**< Co-efficient a of the equation. */
+  const uint32_t *pui32B;     /**< co-efficient b of the equation. */
+  const uint32_t *pui32Gx;    /**< x co-ordinate value of the generator point. */
+  const uint32_t *pui32Gy;    /**< y co-ordinate value of the generator point. */
+} ecc_curve_info_t;
+
+typedef struct {
+  uint32_t       pui32X[12];  /**< Pointer to value of the x co-ordinate. */
+  uint32_t       pui32Y[12];  /**< Pointer to value of the y co-ordinate. */
+} ec_point_t;
+
+/** @} */
+/*---------------------------------------------------------------------------*/
+/** \name ECC functions
+ *  \note Not all sequencer functions are implemented in this driver
+ *        look at the CC2538 manual for a complete list.
+ * @{
+ */
+
+/** \brief Starts ECC Multiplication.
+ *
+ * \param pui32Scalar Pointer to the buffer containing the scalar
+ *        value to be multiplied.
+ * \param ptEcPt Pointer to the structure containing the
+ *        elliptic curve point to be multiplied.  The point should be
+ *        on the given curve.
+ * \param ptCurve Pointer to the structure containing the curve
+ *        info.
+ * \param pui32ResultVector Pointer to the result vector location
+ *        which will be set by this function.
+ * \param process Process to be polled upon completion of the
+ *        operation, or \c NULL
+ *
+ * This function starts the Elliptical curve cryptography (ECC) point
+ * multiplication operation on the EC point and the scalar value.
+ *
+ * \retval PKA_STATUS_SUCCESS if successful in starting the operation.
+ * \retval PKA_STATUS_OPERATION_INPRG if the PKA hw module is busy doing
+ *         some other operation.
+ */
+uint8_t ecc_mul_start(uint32_t *pui32Scalar,
```

Comment:
I think variables names are not allowed to be camelcase as well, I think this was maybe a leftover from the original code.